src/applications/detection/main.cpp

```diff
@@ -52,6 +53,18 @@ using namespace boost::gil;
 namespace bfs = boost::filesystem;
 
 /**
+ * @brief Check if a string is an integer number.
+ * 
+ * @param[in] s The string to check.
+ * @return Return true if the string is an integer number
+ */
+bool isInteger(std::string s)
+{
+  std::regex e("^-?\\d+");
```

Comment:
Don't be pedantic. We're not going to have 10 cameras anytime soon so replace it with `s.size()==1 && std::isdigit(s[0])`.